cohttp-eio/src/body.ml

```diff
@@ -44,29 +45,24 @@ let pp_chunk fmt = function
         fmt chunk
   | Last_chunk extensions -> pp_chunk_extension fmt extensions
 
-open Reader
-open Eio.Buf_read
-
 let read_fixed t headers =
-  match Http.Header.get headers "Content-length" with
-  | Some v ->
-      let content_length = int_of_string v in
-      let content = take content_length t in
-      content
-  | None -> raise @@ Invalid_argument "Request is not a fixed content body"
```

Comment:
If there's no content-length then we should use `Buf_read.take_all` I think.

Reply:
A missing `Content-Length` (if a client wants to send a fixed body request) represents a malformed request. Should `cohttp-eio` compensate for this scenario? I suppose if we don't care about HTTP pipe-lining  then `Buf_read.take_all` would work.

Reply:
Other http clients wait for the end of the connection if there's no length header. That's why the default server responses are all broken at the moment (see #883). Maybe the correct behaviour depends on the HTTP version?

Reply:
Agreed, IIUC #883 and #913 issues are the cohttp-eio servers not behaving correctly by setting the correct content-length header. As such I'd like to handle #883 and #913 in separate PRs after this PR. I would like to keep this PR focused mostly on cohttp Client functionality.

Reply:
The point I'm trying to make is that HTTP responses don't have to have length headers, and we still need to handle that. https://www.rfc-editor.org/rfc/rfc7230 says:

> The length of a message body is determined by one of the following
   (in order of precedence):
> [...]
>  7.  Otherwise, this is a response message without a declared message
>       body length, so the message body length is determined by the
>       number of octets received prior to the server closing the
>       connection.

Reply:
Thanks. This is done now in https://github.com/mirage/ocaml-cohttp/pull/879/commits/073ff8e1a35d7aa9876ee3d521c52c3bd7879dc2.